cmd/kubeadm/app/cmd/reset.go

```diff
@@ -55,99 +56,109 @@ type Reset struct{}
 
 func NewReset(skipPreFlight bool) (*Reset, error) {
 	if !skipPreFlight {
-		fmt.Println("Running pre-flight checks")
-		err := preflight.RunResetCheck()
-		if err != nil {
+		fmt.Println("[preflight] Running pre-flight checks...")
+
+		if err := preflight.RunResetCheck(); err != nil {
 			return nil, &preflight.PreFlightError{Msg: err.Error()}
 		}
 	} else {
-		fmt.Println("Skipping pre-flight checks")
+		fmt.Println("[preflight] Skipping pre-flight checks...")
 	}
 
 	return &Reset{}, nil
 }
 
-// cleanDir removes everything in a directory, but not the directory itself:
-func cleanDir(path string) {
+// Run reverts any changes made to this host by "kubeadm init" or "kubeadm join".
+func (r *Reset) Run(out io.Writer) error {
+	serviceToStop := "kubelet"
+	initSystem, err := initsystem.GetInitSystem()
+	if err != nil {
+		fmt.Printf("[reset] Failed to detect init system and stop the kubelet service: %v\n", err)
+	} else {
+		fmt.Printf("[reset] Stopping the %s service...\n", serviceToStop)
+		if err := initSystem.ServiceStop(serviceToStop); err != nil {
+			fmt.Printf("[reset] Failed to stop the %s service", serviceToStop)
+		}
+	}
+
+	fmt.Println("[reset] Unmounting directories in /var/lib/kubelet...")
+	umountDirsCmd := "cat /proc/mounts | awk '{print $2}' | grep '/var/lib/kubelet' | xargs -r umount"
```

Comment:
maybe instead of embedding "shell" script here, write small function to call umount individually and report error on particular path ?